src/scripts/items/EvolutionStone.ts

```diff
@@ -14,33 +14,40 @@ class EvolutionStone extends Item {
         player.gainItem(GameConstants.StoneType[this.type], n)
     }
 
-
-
     public use(pokemon?:string) {
         let shiny = PokemonFactory.generateShiny(GameConstants.SHINY_CHANCE_STONE);
-        if(pokemon == "Eevee") {
-            switch (this.type) {
+        let evolution = EvolutionStone.computeEvolution(this.type, pokemon);
+        player.capturePokemon(evolution, shiny, false);
+        return shiny;
+    }
+
+    public static computeEvolution(type: GameConstants.StoneType, pokemon: string): string {
+        if (pokemon == "Eevee") {
+            switch (type) {
                 case GameConstants.StoneType.Fire_stone: {
-                    player.capturePokemon("Flareon", shiny, false);
-                    break;
+                    return "Flareon";
                 }
                 case GameConstants.StoneType.Water_stone: {
-                    player.capturePokemon("Vaporeon", shiny, false);
-                    break;
+                    return "Vaporeon";
                 }
                 case GameConstants.StoneType.Thunder_stone: {
-                    player.capturePokemon("Jolteon", shiny, false);
-                    break;
+                    return "Jolteon";
+                }
+            }
+        } else if (pokemon == "Gloom") {
+            switch (type) {
+                case GameConstants.StoneType.Leaf_stone: {
+                    return "Vileplume";
+                }
+                case GameConstants.StoneType.Sun_stone: {
+                    return "Bellossom";
                 }
             }
         }
         else {
-            let evolution: string = PokemonHelper.getPokemonByName(pokemon).evolution;
-            player.capturePokemon(evolution, shiny, false);
+            return PokemonHelper.getPokemonByName(pokemon).evolution;
```

Comment:
Maybe build a check in case this doesn't return anything (i.e. we forgot to hardcode a special evolution).